internal/connect/command_offset_update.go

```diff
@@ -0,0 +1,137 @@
+			},
+		),
+	}
+
+	cmd.Flags().String("config-file", "", "JSON file containing connector offsets to set to.")
+	pcmd.AddClusterFlag(cmd, c.AuthenticatedCLICommand)
+	pcmd.AddContextFlag(cmd, c.CLICommand)
+	pcmd.AddEnvironmentFlag(cmd, c.AuthenticatedCLICommand)
+	pcmd.AddOutputFlag(cmd)
+
+	cobra.CheckErr(cmd.MarkFlagFilename("config-file", "json"))
+	cobra.CheckErr(cmd.MarkFlagRequired("config-file"))
+
+	return cmd
+}
+
+func (c *offsetCommand) update(cmd *cobra.Command, args []string) error {
+	kafkaCluster, err := kafka.GetClusterForCommand(c.V2Client, c.Context)
+	if err != nil {
+		return err
+	}
+
+	environmentId, err := c.Context.EnvironmentId()
+	if err != nil {
+		return err
+	}
+
+	connector, err := c.V2Client.GetConnectorExpansionById(args[0], environmentId, kafkaCluster.ID)
+	if err != nil {
+		return err
+	}
+
+	connectorName := connector.Info.GetName()
+	configFile, err := cmd.Flags().GetString("config-file")
+	if err != nil {
+		return err
+	}
+
+	request, err := c.getAlterOffsetRequestBody(configFile)
+	if err != nil {
+		return err
+	}
+
+	alterOffsetRequestInfo, err := c.V2Client.AlterConnectorOffsets(connectorName, environmentId, kafkaCluster.ID, *request)
+	if err != nil {
+		return err
+	}
+
+	offsetStatus := connectv1.ConnectV1AlterOffsetStatus{
+		Request: alterOffsetRequestInfo,
+		Status: connectv1.ConnectV1AlterOffsetStatusStatus{
+			Phase: "PENDING",
+		},
+	}
+
+	var apiErr error
+	_ = retry.Retry(time.Second, 30*time.Second, func() error {
+		offsetStatus, apiErr = c.V2Client.AlterConnectorOffsetsRequestStatus(connectorName, environmentId, kafkaCluster.ID)
+		if apiErr != nil {
+			return nil
+		}
+
+		if strings.ToUpper(offsetStatus.Status.Phase) != "PENDING" {
+			return nil
+		}
+		return fmt.Errorf("update offset request still pending, checking status again")
+	})
+	if apiErr != nil {
+		return apiErr
+	}
+
+	if strings.ToUpper(offsetStatus.Status.Phase) == "PENDING" {
+		output.Println(false, "Operation is PENDING. Please run `confluent connect offset status describe` command to get the latest status of the update request.")
+		return nil
+	}
+
+	if output.GetFormat(cmd) == output.Human {
+		return printHumanDescribeOffsetStatus(cmd, offsetStatus, args[0])
+	}
+
+	return printSerializedDescribeOffsetStatus(cmd, offsetStatus, args[0])
+}
+
+func (c *offsetCommand) getAlterOffsetRequestBody(configFile string) (*connectv1.ConnectV1AlterOffsetRequest, error) {
+	jsonFile, err := os.ReadFile(configFile)
+	if err != nil {
+		return nil, fmt.Errorf(errors.UnableToReadConfigurationFileErrorMsg, configFile, err)
+	}
+
+	if len(jsonFile) == 0 {
+		return nil, fmt.Errorf(`offset configuration file "%s" is empty`, configFile)
+	}
+
+	var request *connectv1.ConnectV1AlterOffsetRequest
+	if err := json.Unmarshal(jsonFile, &request); err != nil {
+		return nil, fmt.Errorf(errors.UnableToReadConfigurationFileErrorMsg, configFile, err)
+	}
+
+	request.SetType("PATCH")
+
+	return request, err
```

Comment:
`err` will always be `nil` at this point
```suggestion
	return request, nil
```

Reply:
Done